<a href="https://colab.research.google.com/github/MariiaYarmolenko/HW-Data-Loves/blob/main/HW_15_3_%D0%A2%D0%B5%D1%81%D1%82%D0%B8_%D0%B4%D0%BB%D1%8F_%D0%BC%D0%B0%D0%BB%D0%B8%D1%85_%D0%B2%D0%B8%D0%B1%D1%96%D1%80%D0%BE%D0%BA_%D1%82%D0%B0_%D0%BF%D1%80%D0%BE%D0%BF%D0%BE%D1%80%D1%86%D1%96%D0%B9.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Тести для малих вибірок та пропорцій



**Завдання 1**. E-commerce компанія після редизайну сайту підозрює, що **середній час до покупки (time-to-purchase)** користувачів **збільшився**.
Історично середній час від першого заходу на сайт до покупки становив $\mu_0 = 123.7$ хвилин.

Команда аналітиків випадково обрала дані **7 покупців після редизайну**:
`128, 135, 121, 142, 126, 151, 123`

З допомогою наявних даних зʼясуйте, чи збільшився середній час покупки після редизайну.

Для цього виконайте наступні 6 кроків. Правильне виконання кожного з кроків оцінюється в 1 бал.

1. Запишіть нульову та альтернативну гіпотези і визначте тип тесту.
2. Обчисліть вибіркові статистики: $\bar x$, $s$, $n$.
3. Оберіть тип тесту та виконайте його (знайдіть критичне значення тесту, статистику тесту та р-значення) будь-яким пасуючим способом, щоб перевірити гіпотезу на рівні значущості $\alpha = 0.10$.
4. Прийміть рішення, чи відхиляєте ви гіпотезу $H_0$ використовуючи p-value.
5. Напишіть висновок: чи справді редизайн сайту подовжив час до покупки?
6. Чи зміниться ваше рішення при зміні рівня значущості на $\alpha = 0.05$.


In [1]:
import numpy as np
import pandas as pd
from scipy import stats
from statsmodels.stats import weightstats as stests

$H_0$: μ = 123.7

$H_a$: μ > 123.7

Тип тесту: правосторонній

In [44]:
data = [128, 135, 121, 142, 126, 151, 123]
mu_0 = 123.7
alpha = 0.10

x_bar = np.mean(data)
s = np.std(data, ddof=1)
n = len(data)
df = n - 1

t_stat1 = (x_bar - mu_0) / (s / np.sqrt(n))

t_crit1 = stats.t.ppf(1 - alpha, df)

p_value1 = 1 - stats.t.cdf(t_stat1, df)

if p_value1 < alpha:
    print(f"Результат значущий (p = {p_value1:.3f}).")
    print(f"Відхиляємо H0: є підстави вважати, що середній час зріс.")
else:
    print(f"Результат не значущий (p = {p_value1:.3f}).")
    print("Не відхиляємо H0: доказів зростання часу недостатньо.")

Результат значущий (p = 0.042).
Відхиляємо H0: є підстави вважати, що середній час зріс.


## Відхилення нульової гіпотези вказує на те, що редизайн дійсно подовжив час до покупки.

## Зміна α на 0.05 не вплине на результат, оскільки 0.042 < 0.05, але різниця вже не така велика, як з α 0.1

**Завдання 2.**

До спеціальної рекламної кампанії **23%** дорослих упізнавали логотип компанії. Після завершення кампанії відділ маркетингу провів опитування: з **1200** випадково відібраних дорослих **311** упізнали логотип.

Перевірте на рівні значущості **3%** ($\alpha=0.03$), чи дають ці дані достатні підстави стверджувати, що **тепер більше ніж 23%** дорослих упізнають логотип компанії. Для розвʼязку використовуйте бібліотеку `statsmodels`.

Зробіть висновок, чи зросла впізнаваність логотипу.

Додатково, обчисліть довірчий інтревал на заданому рівні значущості і проінтерпретуйте текстом - як він додатково пояснює прийняте нами рішення?

In [46]:
import statsmodels.api as sm
from statsmodels.stats.proportion import proportions_ztest, proportion_confint

count = 311
nobs = 1200
p0 = 0.23

stat, pval = proportions_ztest(count, nobs, p0, alternative='larger')
confint = proportion_confint(count, nobs, alpha=0.03, method='normal')

print("🔹 Тест для однієї пропорції")
print(f"Z-статистика = {stat:.3f}")
print(f"p-value = {pval:.4f}")
print(f"Довірчий інтервал: {confint}")

🔹 Тест для однієї пропорції
Z-статистика = 2.306
p-value = 0.0106
95% довірчий інтервал: (0.23171700302179205, 0.28661633031154127)


H0:p=0.23

H1:p>0.23

Оскільки  p-value = 0.0106 < 0.03 , відхиляємо  H0 .

Є статистично значущі докази, що частка тих, хто впізнає цей бренд, більша за 23%.

97% довірчий інтервал (0.232, 0.289)

Це означає, що з імовірністю 97% справжня частка лежить у межах від 23.2% до 28.9%.